airflow/configuration.py

```diff
@@ -681,18 +674,78 @@ def remove_option(section, option):
     return conf.remove_option(section, option)
 
 
+def set(section, option, value):  # noqa
+    return conf.set(section, option, value)
+
+
+########################
+# Convenience method to access config entries.
+
+
+def get_dags_folder():
+    return os.path.expanduser(get('core', 'DAGS_FOLDER'))
+
+
+def get_airflow_home():
+    return os.path.expanduser(get('core', 'AIRFLOW_HOME'))
+
+
+def get_sql_alchemy_conn():
+    return get('core', 'SQL_ALCHEMY_CONN')
+
+
+################
+# global config init
+
+conf = None
+
+
+def test_mode():
+    conf = ConfigParserWithDefaults(defaults)
+    conf.read(TEST_CONFIG_FILE)
+
+
+def load_config():
+    """
+    loads the config and triggers the connection to the SQL-alchemy backend
+    """
+    global conf
+    conf = ConfigParserWithDefaults(defaults)
+    conf.read(AIRFLOW_CONFIG)
+    settings.connect(get_sql_alchemy_conn(),
+                     conf.getint('core', 'SQL_ALCHEMY_POOL_SIZE'),
+                     conf.getint('core', 'SQL_ALCHEMY_POOL_RECYCLE'))
+
+load_config()
```

Comment:
Create load_config() that loads the config from AIRFLOW_CONFIG and calls settings.py to open the sql alchemy connection. This is useful to reload the config when the AIRFLOW_CONFIG will be modified during tests.